### PR TITLE
Command history

### PR DESCRIPTION
**Pull request by YowSiaoKang**

### CODE REVIEW

**Comment by YowSiaoKang** [comment]
Known Issue:
- input caret disappears eventhough textfield in focus -> issue with ComboBox component in javaFX -> fixes: try pressing "Tab" a few times for caret to appear upon refocus

File: src/main/java/seedu/address/ui/AutoCompleteComboBoxPlus.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            commandText = commandText.trim();
+            if (commandHistory.contains(commandText)) {
+                commandHistory.remove(commandText);
+            }
+            commandHistory.add(0, (T) commandText);
+            comboBox.setItems(commandHistory);
+            comboBox.hide();
+            return;
+        } catch (CommandException | ParseException e) {
+            setStyleToIndicateCommandFailure(comboBox);
```
Comment (Benson15912) [inline src/main/java/seedu/address/ui/AutoCompleteComboBoxPlus.java:170]: After showing the error, i think its better to hide the combobox again by adding the line:
comboBox.hide();

**Review by Benson15912 — APPROVED** [review]
LGTM

File: src/main/resources/view/MainWindow.fxml
Comment (xuwen-ho) [inline src/main/resources/view/MainWindow.fxml:1]: Should we put the whole match logo here instead?

File: src/main/java/seedu/address/ui/AutoCompleteComboBoxPlus.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            comboBox.hide();
+            return;
+        }
+    }
+
+    private static <T> void processTextInput(ComboBox<T> comboBox, ObservableList<T> commandHistory,
```
Comment (xuwen-ho) [inline src/main/java/seedu/address/ui/AutoCompleteComboBoxPlus.java:174]: Could this method be shortened maybe by splitting up into eg. processTextInput, filterList and updateComboBox methods?

File: src/main/java/seedu/address/ui/AutoCompleteComboBoxPlus.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        // Handle the key typed event properly
+        comboBox.addEventHandler(KeyEvent.KEY_PRESSED, t -> comboBox.hide());
+        comboBox.addEventHandler(KeyEvent.KEY_RELEASED,
+                                 createHandler(comboBox, comparatorMethod, commandHistory, commandExecutor));
+    }
+
+    /**
+     * Creates a EventHandler for the ComboBox.
+     */
+    private static <T> EventHandler<KeyEvent> createHandler(ComboBox<T> comboBox,
```
Comment (xuwen-ho) [inline src/main/java/seedu/address/ui/AutoCompleteComboBoxPlus.java:57]: Code is well organised and readable!

**Review by xuwen-ho — APPROVED** [review]
Function works great, offers suggestions when keying in commands and also remembers previous commands.

**Pull request merged**